provider/lxd/provider.go

```diff
@@ -236,40 +245,22 @@ func (p *environProvider) DetectClouds() ([]cloud.Cloud, error) {
 			})
 		}
 	}
-
 	return clouds, nil
 }
 
 // DetectCloud implements environs.CloudDetector.
 func (p *environProvider) DetectCloud(name string) (cloud.Cloud, error) {
-	// For now we just return a hard-coded "localhost" cloud,
-	// i.e. the local LXD daemon. We may later want to detect
-	// locally-configured remotes.
-	switch name {
-	case lxdnames.ProviderType, lxdnames.DefaultCloud:
-		return localhostCloud, nil
-	default:
-		configPath := filepath.Join(utils.Home(), ".config", "lxc", "config.yml")
-		config, err := p.lxcConfigReader.ReadConfig(configPath)
-		if err != nil {
-			logger.Errorf("unable to read LXC config file %s", err)
-			break
-		}
-
-		if remote, ok := config.Remotes[name]; ok {
-			return cloud.Cloud{
-				Name:        name,
-				Type:        lxdnames.ProviderType,
-				Endpoint:    remote.Addr,
-				Description: "LXD Cluster",
-				AuthTypes: []cloud.AuthType{
-					cloud.CertificateAuthType,
-				},
-				Regions: []cloud.Region{{
-					Name:     lxdnames.DefaultRemoteRegion,
-					Endpoint: remote.Addr,
-				}},
-			}, nil
+	clouds, err := p.DetectClouds()
+	if err != nil {
+		return cloud.Cloud{}, errors.Trace(err)
+	}
+	match := name
+	if match == lxdnames.DefaultCloudAltName {
+		match = lxdnames.DefaultCloud
+	}
```

Comment:
I couldn't see a test for this (may have missed it). But would be good to have one